### ISSUE TITLE
Should plot with no layers add geom_blank() ?

### ISSUE THREAD

**Issue by hadley** [hadley/ggplot2#1246]
Instead of current (not very useful behaviour)

**Comment by wch** [comment]
I think this would be useful in Shiny apps, where layers are added in response to some user input. Right now, if no layers are added, you'd need to do an explicit check for that, and then add `geom_blank()`. If this automatically happens, then that's less work for the developer, and there's not really any drawbacks that I can think of.

This would happen in the build phase, right?

**Comment by hadley** [comment]
Yeah, I'd just change:

```R
if (length(plot$layers) == 0) stop("No layers in plot", call. = FALSE)
``` 

to 

```R
if (length(plot$layers) == 0) {
  plot <- plot + geom_blank()
}
```

**Comment by hadley** [comment]
Fixed in 1ab70defc61064a5f39121341f77c35d1a061518
